### DISCUSSION TITLE
Exception handling

### DISCUSSION THREAD

**Discussion by iammrsea** [adonisjs/core#1533] [Help]
Hello guys,

Please any ideas on how to properly handle exceptions when my app is both an api server
for mobile app and a web application. It's kind of confusing to me on how to make it return
appropriate error pages for the web app and json responses for the api.

**Comment by thetutlage** [comment]
Use content negotation. The docs already talk about Validator using content negotiation to choose the appropriate error reporter https://preview.adonisjs.com/guides/validator/error-reporters#content-negotiation. The same can be applied to the entire app. You will have to read about content negotation separately, since it is a widely used concept
